aten/src/ATen/core/boxing/KernelFunction_impl.h

```diff
@@ -43,16 +50,46 @@ inline Return callUnboxedKernelFunction(void* unboxed_kernel_func, OperatorKerne
     return (*func)(functor, dispatchKeySet, std::forward<Args>(args)...);
 }
 
+template <typename T>
+inline T unpackSymInt(T x) { return x; }
+
+inline int64_t unpackSymInt(c10::SymInt x) {
+  return x.expect_int();
+}
+
+inline c10::IntArrayRef unpackSymInt(c10::SymIntArrayRef x) {
+  return c10::asIntArrayRefSlow(x);
+}
+
+inline c10::optional<int64_t> unpackSymInt(c10::optional<c10::SymInt> x) {
+  return x.has_value() ? c10::make_optional(x->expect_int()) : c10::nullopt;
+}
+
 template<class Return, class... Args>
 C10_ALWAYS_INLINE Return KernelFunction::call(const OperatorHandle& opHandle, DispatchKeySet dispatchKeySet, Args... args) const {
     // note: Args above is intentionally not Args&&. We don't want perfect
     // forwarding, which would require Args to be deduced, but instead we
     // want callers to explicitly specify the Args.
 
-    if (C10_LIKELY(unboxed_kernel_func_ != nullptr)) {
-      auto *functor = boxed_kernel_func_.getFunctor();
-      return callUnboxedKernelFunction<Return, Args...>(
-          unboxed_kernel_func_, functor, dispatchKeySet, std::forward<Args>(args)...);
+    // This should get inlined by compiler
+    if (guts::disjunction<has_symint<Args>...>::value) {
+      if (sym_unboxed_kernel_func_ != nullptr) {
+        auto *functor = boxed_kernel_func_.getFunctor();
```

Comment:
is the getFunctor call so we can get some python state associated with the boxed kernel and then use that same state with the sym_unboxed kernel?  I'm pretty unfamiliar here

Reply:
No, this is a twisty corner of the boxing implementation. Functor refers to a rarely used piece of functionality in KernelFunction which allows operators to be defined as arbitrary Functor classes that can have extra state in them (much like how a lambda that closes over variables has extra state). Closure-less lambdas and regular function pointers are functors in a degenerate sense, they have no state. Everything goes through functor, so I updated the functor path only. It doesn't really matter, just mentally substitute functor with function and the code is still meaningful.